functions.py

```diff
@@ -109,25 +108,112 @@ def sumout(f: Factor, v: str) -> Factor:
 
 def normalize(f: Factor) -> Factor:
     """
-    function that normalizes a factor by dividing each entry by the sum of all the entries. 
+    function that normalizes a factor by dividing each entry by the sum of all the entries.
     This is useful when the factor is a distribution (i.e. sum of the probabilities must be1).
     """
     the_sum = sum(f.data.values())
     for key in f.kit:
         f.data[key] = f.data[key]/the_sum
-    assert(sum(f.data.values()) == 1.0)
+    assert(sum(f.data.values()) > 0.999999 and sum(f.data.values()) < 1.000001)
```

Comment:
I had to do this because the assertion was failing randomly. Remember how I said that it's neat how python allows you to compare equality with floats? I guess that's not always true 🤔

Reply:
Huh, that's wild but nice catch